### ISSUE TITLE
rnn.py no longer breaks as expected when adding units

### ISSUE THREAD

**Issue by charlesfrye** [lukas/ml-class#62]
In the [RNN prediction video](https://youtu.be/8lbGjKhrJOo?t=930), a shape error is generated when the RNN outputs a 5-element hidden state when it should be outputting a 1-element hidden state.

Weirdly enough, in a more recent version of Keras, this error no longer occurs.

I suspect some weirdness is happening inside the loss function calculation.

Interestingly, the predictions on the validation data no longer line up, as can be seen in [this project](https://app.wandb.ai/charlesfrye/ml-class/runs/1itn27d1?workspace=user-charlesfrye).
